### ISSUE TITLE
Potentially wrong grid size in `shr_dmodel_readgrid` for multi-column scm

### ISSUE THREAD

**Issue by mwarusz** [E3SM-Project/E3SM#6570]
I am not familiar with this code at all, but this line looks suspect to me and it caused overflows in some of my DP-SCREAM runs:

https://github.com/E3SM-Project/E3SM/blob/f50d2639c68c1607a521407b420af040080c1380/share/streams/shr_dmodel_mod.F90#L386

Shouldn't it be `scm_nx * scm_ny * nzgeo` ?

**Comment by bogensch** [comment]
Thanks for reporting.  So that I can reproduce the error could you please provide some more information for a setup where you see this error.  Such as machine used, case, number of elements in the x and y directions and as well domain size.  Providing the exact script you used would be best.

**Comment by ambrad** [comment]
Excellent find, Maciej! I'm investigating another issue and have come across this same line as a problem. @bogensch it's possible that the GNU issues you see on PM are in fact due to this line. I have a reliable reproducer on Chrysalis with Intel running and GNU failing.

**Comment by bogensch** [comment]
Yes thanks @mwarusz and @ambrad, I will test this fix first chance I get.  I must admit, I reread this issue incorrectly the first time and I see now the error is pretty obvious.

**Comment by bogensch** [comment]
I can report that putting this fix in solves all the issues I was having when attempting large domain sizes with DPxx.  thanks for reporting @mwarusz! I will issue a PR for this fix early next week.
